ydb/core/persqueue/pq_impl.cpp

```diff
@@ -4424,6 +4431,30 @@ void TPersQueue::Handle(TEvPQ::TEvPartitionScaleStatusChanged::TPtr& ev, const T
     }
 }
 
+void TPersQueue::Handle(TEvPQ::TEvDeletePartitionDone::TPtr& ev, const TActorContext& ctx)
+{
+    auto* event = ev->Get();
+    const ui64 writeId = event->Cookie;
+    Y_ABORT_UNLESS(TxWrites.contains(writeId));
+    TTxWriteInfo& writeInfo = TxWrites.at(writeId);
+    Y_ABORT_UNLESS(writeInfo.Partitions.contains(event->PartitionId.OriginalPartitionId));
+    const TPartitionId partitionId = writeInfo.Partitions.at(event->PartitionId.OriginalPartitionId);
+    Y_ABORT_UNLESS(partitionId == event->PartitionId);
+    Y_ABORT_UNLESS(partitionId.IsSupportivePartition());
+    Y_ABORT_UNLESS(Partitions.contains(partitionId));
+    const TPartitionInfo& partition = Partitions.at(partitionId);
+
+    Send(partition.Actor, new TEvents::TEvPoisonPill());
```

Comment:
Call die() or PassAway() in partition actor on sending EvDeletePartitionDone.

Reply:
Done
https://github.com/ydb-platform/ydb/pull/4407/commits/1b4b028a3da95418b80615b2e6bd7676b39d4158